src/hardware/mixer.cpp

```diff
@@ -117,33 +121,71 @@ static struct mixer_t mixer = {};
 
 uint8_t MixTemp[MIXER_BUFSIZE] = {};
 
-MixerChannel::MixerChannel(MIXER_Handler _handler, const char *_name)
+MixerChannel::MixerChannel(MIXER_Handler _handler, const char *_name,
+                           const std::set<ChannelFeature> &_features)
         : name(_name),
           envelope(_name),
-          handler(_handler)
+          handler(_handler),
+          features(_features)
 {}
 
+bool MixerChannel::HasFeature(ChannelFeature feature)
+{
+	return features.find(feature) != features.end();
+}
+
 bool MixerChannel::StereoLine::operator==(const StereoLine &other) const
 {
 	return left == other.left && right == other.right;
 }
 
-mixer_channel_t MIXER_AddChannel(MIXER_Handler handler, const int freq, const char *name)
+static void set_global_channel_settings(mixer_channel_t channel)
+{
+	const auto sect = static_cast<Section_prop *>(control->GetSection("mixer"));
+	assert(sect);
+
+	// Global crossfeed
+	auto crossfeed = 0.0f;
+
+	const std::string crossfeed_pref = sect->Get_string("crossfeed");
+	if (crossfeed_pref == "on") {
+		constexpr auto default_crossfeed_strength = 0.3f;
+		crossfeed = default_crossfeed_strength;
+	} else if (crossfeed_pref == "off") {
+		crossfeed = 0.0f;
+	} else {
+		const auto cf = to_finite<double>(crossfeed_pref);
+		if (std::isfinite(cf) && cf >= 0.0 && cf <= 100.0) {
+			crossfeed = cf / 100.0f;
+		} else {
+			LOG_WARNING("MIXER: Invalid crossfeed value '%s', using off",
+			            crossfeed_pref.c_str());
```

Comment:
This is working nicely. Tried lots of out of range and illegal values. No sweat handling them all and behaving as I expect :+1: